packages/babel-generator/src/generators/jsx.js

```diff
@@ -43,7 +44,8 @@ export function JSXSpreadChild(node: Object) {
 }
 
 export function JSXText(node: Object) {
-  this.token(node.value);
+  const raw = this.getPossibleRaw(node);
+  this.token(raw || node.value);
```

Comment:
Can we explicitly check `=== null` since `raw` being an empty string would also cause this to use `.value` instead?

Thinking on this all a bit, I'm up for adding `raw` handling if we _need_ it, but it seems like we should also have additional logic both here and in StringLiteral to handle encoding of `.value`, otherwise it's still totally possible for people to assign values to JSXText that will not be printed properly. The intention of `.raw` is to not lose the original formatting in the source file, if possible, but either way we don't want usage of `.value` without `.raw` to cause issues.